auth/grpctransport/grpctransport.go

```diff
@@ -130,7 +130,7 @@ func (o *Options) resolveDetectOptions() *credentials.DetectOptions {
 	do := transport.CloneDetectOptions(o.DetectOpts)
 
 	// If scoped JWTs are enabled user provided an aud, allow self-signed JWT.
-	if (io != nil && io.EnableJWTWithScope) || do.Audience != "" {
+	if (io != nil && io.EnableJWTWithScope) || do.Audience != "" || !o.isUniverseDomainGDU() {
```

Comment:
It is not clear to me this is the desired behavior. If a scope or auth is not present you can't really make a valid self-signed JWT. Instead of saying if nonGDU assume self-signed, maybe we should validate that it will be a self-signed JWT?

Reply:
yeah, to me that looks flawed too. I think we could make this change. But does it fix any bug? If it does fix a real use case I am fine with it, but it is not clear to me that it does.

Reply:
I'll try to move this logic to a later stage at which 2) service account can be determined.

Reply:
Even if it is a service account, you need a scope or an aud for the JWT. Do you have a use-case to test this against?

Reply:
Can the`InternalOptions.DefaultScopes` be used?

Reply:
Only if jwtscope is enabled. Iirc not all apis support scoped JWTs. Aud JWTs work for anything though I believe.

Reply:
I'll see if I can get some info about how self-signed JWTs are intended to be used with universe domain.

Reply:
My understanding is that TPC participating services must support self-signed JWTs, and my best guess is that the default scope(s) are good enough.